### PR TITLE
Extract strings to constants

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

Some polishing, there are multiples places where these strings are used, this replaces the scalar strings by constants.

Since they seem related to `Datagrid`, I've created them in `DatagridInterface`. ~I've chosen `KEY_` prefix, but any other name is welcome.~

I am targeting this branch, because these changes are BC.

## Changelog

```markdown
### Added
- Added `DatagridInterface::SORT_ORDER` constant to use it instead of `_sort_order` string.
- Added `DatagridInterface::SORT_BY` constant to use it instead of `_sort_by` string.
- Added `DatagridInterface::PAGE` constant to use it instead of `_page` string.
- Added `DatagridInterface::PER_PAGE` constant to use it instead of `_per_page` string.
```

### CODE REVIEW

**Review by VincentLanglet — APPROVED** [review]
Seems nice

**Comment by jordisala1991** [comment]
I think we are using as a key of an array is an implementation detail, what about just removing the prefix?

File: docs/cookbook/recipe_sortable_listing.rst [hunk 1]
```diff
@@ -117,16 +117,17 @@ Now we need to define the sort by field to be ``$position``::
...
 
     final class ClientAdmin extends AbstractAdmin
     {
         protected function configureDefaultSortValues(array &$sortValues): void
         {
-            $sortValues['_page'] = 1;
-            $sortValues['_sort_order'] = 'ASC';
-            $sortValues['_sort_by'] = 'position';
+            $sortValues[DatagridInterface::PAGE] = 1;
+            $sortValues[DatagridInterface::SORT_ORDER] = 'ASC';
```
Comment (core23) [inline docs/cookbook/recipe_sortable_listing.rst:129]: You can use https://github.com/doctrine/collections/blob/1.6.x/lib/Doctrine/Common/Collections/Criteria.php#L18 for `ASC` and `DESC`

Comment (franmomu) [reply docs/cookbook/recipe_sortable_listing.rst:129]: Not sure about this, the phpdoc of the class says:

```php
/**
 * Criteria for filtering Selectable collections.
 */
```
and the usage inside the `doctrine` organization: https://github.com/search?q=org%3Adoctrine+Criteria%3A%3ADESC&type=Code looks like it's only around the `Criteria` class.

**Review by VincentLanglet — APPROVED** [review]

**Review by phansys — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Review by core23 — APPROVED** [review]

**Comment by jordisala1991** [comment]
Thnk you @franmomu

**Pull request merged**